designer/restmodel/src/main/scala/pl/touk/nussknacker/restmodel/displayedgraph/DisplayableProcess.scala

```diff
@@ -50,21 +53,68 @@ import pl.touk.nussknacker.engine.graph.NodeDataCodec._
 }
 
 @JsonCodec(decodeOnly = true)
-case class ProcessProperties(typeSpecificProperties: TypeSpecificData,
-                             additionalFields: Option[ProcessAdditionalFields] = None) {
+case class ProcessProperties(additionalFields: ProcessAdditionalFields, propertiesType: String) {
+
+  def toMetaData(id: String): MetaData = {
+
+    val (typeSpecificData, additionalFields) = extractTypedData(this)
 
-  def toMetaData(id: String): MetaData = MetaData(
-    id = id,
-    typeSpecificData = typeSpecificProperties,
-    additionalFields = additionalFields
-  )
-  val isSubprocess: Boolean = typeSpecificProperties.isSubprocess
+    MetaData(
+      id = id,
+      typeSpecificData = typeSpecificData,
+      additionalFields = additionalFields
+    )
+  }
+
+  val isSubprocess: Boolean = extractTypedData(this)._1.isSubprocess
 
 }
 
 object ProcessProperties {
-  implicit val encodeProcessProperties: Encoder[ProcessProperties] =
-    Encoder.forProduct3("typeSpecificProperties", "isSubprocess", "additionalFields") { p =>
-    (p.typeSpecificProperties, p.isSubprocess, p.additionalFields)
+
+  def apply(typeSpecificProperties: TypeSpecificData,
+            additionalFields: Option[ProcessAdditionalFields] = None): ProcessProperties = {
+    val genericTypeSpecificData = typeSpecificProperties.toGenericMap
+    val additionalProps = additionalFields.map(a => a.properties).getOrElse(None)
+    val description = additionalFields.flatMap(a => a.description)
+
+    val props = ProcessAdditionalFields(description, genericTypeSpecificData ++ additionalProps)
```

Comment:
what happens when we have duplicate propertyNames? (I mean one in additonal and one in typespecific)